### PR TITLE
Fixed causality loop detection bug

### LINKED ISSUE #1024: Dependency cycle detection does not seem to work
Hi, tried to do the esweek tutorial and I noticed that the dependency cycle analysis does not work.

```
target C

reactor Printer {
    input in: int
    output out: int
    reaction(in) -> out {=
        // Using a thread-safe print function provided by the runtime.
        info_print("Hello World! %d", in->value);
    =}
}

main reactor {
    p = new Printer()
    p.out -> p.in
}
```
tutorial

No validator error is created. So it does not seem to be a problem of the diagram.

### PR DESCRIPTION
**Pull request by edwardalee**
This fixes #1024 and updates a test which had a flaw masked by this bug.

### CODE REVIEW

**Review by a-sr — APPROVED** [review]
ModalCycleBreaker test still works as intended.
Thanks for fixing it.

**Comment by edwardalee** [comment]
There is a test failure in the C++ benchmarks, but I can't replicate this. The output from CI seems to imply the failure is in parallelism/MatMul.lf.  I cloned the benchmarks repo, but I can't compile this benchmark. I get the following...  Help?
```
> build-lfc -r src/parallelism/MatMul.lf 
Jar file is missing or out-of-date; starting rebuild...
...
BUILD SUCCESSFUL in 10s
13 actionable tasks: 9 executed, 4 up-to-date
Running lfc...
Generating code for: file:/Users/eal/git/benchmarks-lingua-franca/Cpp/Savina/src/parallelism/MatMul.lf
******** mode: STANDALONE
******** generated sources: /Users/eal/git/benchmarks-lingua-franca/Cpp/Savina/src-gen/parallelism/MatMul
Exception in thread "main" java.io.IOException: A required target resource could not be found: /lib/cpp/reactor-cpp
Perhaps a git submodule is missing or not up to date.
See https://github.com/icyphy/lingua-franca/wiki/downloading-and-building#clone-the-lingua-franca-repository.
Also try to refresh and clean the project explorer if working from eclipse.
	at org.lflang.util.FileUtil.copyDirectoryFromClassPath(FileUtil.java:265)
	at org.lflang.generator.cpp.CppGenerator.generateFiles(CppGenerator.kt:142)
	at org.lflang.generator.cpp.CppGenerator.doGenerate(CppGenerator.kt:77)
	at org.lflang.generator.LFGenerator.doGenerate(LFGenerator.java:164)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:44)
	at org.eclipse.xtext.generator.GeneratorDelegate.generate(GeneratorDelegate.java:35)
	at org.lflang.lfc.Main.runGenerator(Main.java:293)
	at org.lflang.lfc.Main.main(Main.java:222)
```

**Comment by lhstrh** [comment]
Try `git submodule update --init --recursive` in the benchmarks repo?

**Comment by Soroosh129** [comment]
It looks like the test is failing because there is a cycle in FilterBank:
```
[INFO] - lfc: error: Reaction triggers involved in cyclic dependency in reactor Delay: inValue.
[INFO] -   --> src/parallelism/FilterBank.lf:150:14
[INFO] -     |
[INFO] - 149 |
[INFO] - 150 |     reaction(inValue) -> outValue {=
[INFO] -     |              ^^^^^^^ Reaction triggers involved in cyclic dependency in reactor Delay: inValue.
[INFO] -     |
[INFO] - 151 |         double result = *inValue.get();
[INFO] - 
```

File: test/C/src/modal_models/ModalCycleBreaker.lf [hunk 1]
```diff
@@ -18,8 +18,8 @@ reactor Modal {
     mode Two { // Defining reaction to in2 before in1 would cause cycle if no mode were present
         timer wait(150msec, 1sec)
               
-        reaction(in2) -> out {=
-            SET(out, in2->value);
+        reaction(in2) {=
+            // SET(out, in2->value);
```
Comment (lhstrh) [inline test/C/src/modal_models/ModalCycleBreaker.lf:22]: Just tagging @a-sr to be sure he is aware.

**Review by lhstrh — APPROVED** [review]

**Pull request merged**